simbev/simbevMiD.py

```diff
@@ -1141,3 +1146,109 @@ def fast_charging_capacity(
         fast_charging_capacity = 350
 
     return fast_charging_capacity
+
+# Definition of variable:
+# temperature_carinside: temperature
+# energie use cooling/heating: KWh verbraucht pro Einheit??
+
+
+def total_consumption(temperature, timesteps, temperature_carinside, con, distance):
+    """ this function calculates the the consumption of the car and the energy loses to
+    cooling and heating
+
+    temperature: float the temperature for the timestep
+    timesteps: int timesteps for the trip
+    temperature_carinside: int
+    con: float consumption of the car
+    distance: the distance of the drive
+    return: consumption
+    """
+    energy_use_cooling = 0.01575
+    energy_use_heating = 0.025
+    temperature_diff = temperature_carinside - temperature
+
+    if temperature_diff > 0:
+        extra_kwh = temperature_diff * energy_use_heating * timesteps
+    else:
+        extra_kwh = (temperature_diff * -1) * energy_use_cooling * timesteps
+    consumption = distance * con + extra_kwh
+    return consumption
+
+
+def temperture_adapting(start_date, end_date):
+    """ This function provides a temperature list which will be needed for further
+    calculations concerning the loses to heating and cooling in a car.
+
+    start_date: str Start date of simulation
+    end_date: str End date of simulation
+    return: temperature_list list with different temperatures
+    """
+    # read in csv
+    temperature = pd.read_csv("data/Wetterdaten_Berlin.csv", sep=";", skiprows=1)
+    temperature = temperature[['date', 'tavg']]
+
+    # change format of start/end_date to same format as csv
+    [year, month, day] = start_date.split('-')
```

Comment:
anything from this point onward could be replaced if for every trip the timestep or date where it starts gets matched with a row/value of the weather pandas.